### PR TITLE
feat: expander init

### PR DESCRIPTION
**Pull request by letsbelopez**
*Issue #, if available:*

*Description of changes:*
Adds a new expander renderer under view renderer

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: packages/test-generator/lib/generators/NodeTestGenerator.ts [hunk 1]
```diff
@@ -44,6 +46,8 @@ export class NodeTestGenerator extends TestGenerator {
 
   private readonly formRendererFactory: any;
```
Comment (cwoolum) [inline packages/test-generator/lib/generators/NodeTestGenerator.ts:47]: Why are these `any`?

**Review by cwoolum — DISMISSED** [review]
LGTM

File: packages/test-generator/lib/generators/NodeTestGenerator.ts (hunk 1 above)
Comment (letsbelopez) [reply packages/test-generator/lib/generators/NodeTestGenerator.ts:47]: These can be updated to something like this for views. I can update on a follow up. I think it's just leftover from an earlier time and can be updated now.
```
private readonly viewRendererFactory: StudioTemplateRendererFactory<unknown, StudioView, FrameworkOutputManager<unknown>, RenderTextComponentResponse, AmplifyViewRenderer>;
```

File: packages/codegen-ui/lib/types/view/collection.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  };
+}
+
+export interface CollectionBodyContent {
+  bindingProperty?: CollectionBindingProperty;
+  componentSlot?: CollectionBodyComponentSlot;
+}
+
+export interface CollectionBindingTitle {
+  type: 'amplify_binding';
```
Comment (SwaySway) [inline packages/codegen-ui/lib/types/view/collection.ts:34]: can we use the naming pattern as previous binding types?

https://github.com/aws-amplify/amplify-codegen-ui/blob/1c2e3170863303d8ec5623cd9582f9f032493a87/packages/codegen-ui/lib/types/bindings.ts#L93

File: packages/codegen-ui/lib/types/view/collection.ts [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+  content: {
+    bindingProperty: CollectionBindingProperty;
+  };
+}
+
+export interface CollectionComponentSlotBody {
+  type: 'amplify_component_slot';
```
Comment (SwaySway) [inline packages/codegen-ui/lib/types/view/collection.ts:41]: same here with binding format
```suggestion
  type: 'Amplify.ComponentSlot';
```

File: packages/codegen-ui/lib/types/view/view.ts [hunk 4]
```diff
@@ -29,6 +30,8 @@ export interface StudioView {
   viewConfiguration: ViewConfiguration;
 }
 
+export declare type ViewType = 'Table' | 'collection';
```
Comment (SwaySway) [inline packages/codegen-ui/lib/types/view/view.ts:33]: capital case Collection?

File: packages/codegen-ui/lib/types/view/collection.ts (hunk 2 above)
Comment (letsbelopez) [reply packages/codegen-ui/lib/types/view/collection.ts:34]: It's an enum now. You suggested all lowercase on enum values.

File: packages/codegen-ui/lib/types/view/collection.ts (hunk 3 above)
Comment (letsbelopez) [reply packages/codegen-ui/lib/types/view/collection.ts:41]: This is all the result of having a wide variety of enum formats

File: packages/codegen-ui/lib/types/view/view.ts (hunk 4 above)
Comment (letsbelopez) [reply packages/codegen-ui/lib/types/view/view.ts:33]: see above replies. How do we handle it? We'll have to change 'Table' to 'table' eventually.

**Review by cwoolum — APPROVED** [review]
LGTM

**Review by SwaySway — DISMISSED** [review]

**Pull request closed without merging**